src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2022,14 +2020,6 @@ void MergeTreeData::loadDataParts(bool skip_sanity_checks, std::optional<std::un
         }
     }
 
-    if (num_parts == 0 && unexpected_parts_to_load.empty())
-    {
-        resetObjectColumnsFromActiveParts(part_lock);
-        resetSerializationHints(part_lock);
-        LOG_DEBUG(log, "There are no data parts");
-        return;
-    }
```

Comment:
Maybe we should keep this check for non-read-only tables?

Reply:
IIRC, I've unified it, so there is no premature `return` path.